odl/tomo/backends/util.py

```diff
@@ -4,18 +4,17 @@
 def _add_default_complex_impl(fn):
     """Wrapper to call a class method twice when it is complex."""
 
-    def wrapper(self, x, out, **kwargs):
+    def wrapper(self, x, out=None, **kwargs):
         if self.vol_space.is_real and self.proj_space.is_real:
-            fn(self, x, out, **kwargs)
-            return out
+            return fn(self, x, out, **kwargs)
         elif self.vol_space.is_complex and self.proj_space.is_complex:
             result_parts = [
                 fn(self, x.real, getattr(out, 'real', None), **kwargs),
                 fn(self, x.imag, getattr(out, 'imag', None), **kwargs)
             ]
 
             if out is None:
-                out = range.element()
+                out = self.proj_space.element()
```

Comment:
I need to fix this, maybe write a test to cover this?

Reply:
Yes, that's a good idea.

Reply:
Fixed this and updated the complex test to cover this.